### PR TITLE
Updates to P3 scheme

### PR DESCRIPTION
**Pull request by hmorrison100**
TYPE: enhancement and bug fix 

KEYWORDS: P3, microphysics

SOURCE: Hugh Morrison (NCAR MMM)

DESCRIPTION OF CHANGES:
SUMMARY: Updates to improve the P3 scheme (re-factor 3-moment lookup table and minor bug fixes)

1. Reduce 3-moment lookup table file size by an order of magnitude. Re-factoring the lookup table 
dimensions for 3-moment greatly reduces file size with little loss of accuracy. Re-factoring of the 
3-moment lookup table so the dimension is in the space of mu (gamma size distribution shape parameter) 
instead of normalized 6th moment. This reduces file size by an order of magnitude. Unzipping of the file in 
the primary Makefile is no longer needed (file in run/ is no longer gzipped).

2. Improve flexibility by including Kogan (2013, JAS) option for cloud autoconversion/accretion. Addition of 
Kogan (2013) as an option for warm rain (set by iparam=4).

3. Minor bug fixes. 
   * Set rain-ice collection parameter to -99 when rain is not present, instead of 0 (since this is in log space) 
     --> non-answer changing
   * Fix to the double log bug for ice-rain collection
   * Minor LT2 fixes --> replace a few instances of cgp1 with cgp2, and array cgp1 and cgp2 in both i_Fr and 
      i_rhor instead of just rho_r.
   * Update to 2 of the 3 lookup table files and offline lookup table creation codes and corresponding changes 
      to the lookup table filenames in primary Makefile. 

LIST OF MODIFIED FILES: 
M phys/module_mp_p3.F
M ./Makefile 
M run/p3_lookupTable_1.dat-3momI_v5.1.7 --> lookup table file
M run/p3_lookupTable_2.dat-2momI_v5.2.3 --> lookup table file
M run/create_p3_lookupTable_1-2mom_v5.3.f90 --> offline lookup table generation code
M run/create_p3_lookupTable_1-3mom_v5.1.7.f90 --> offline lookup table generation code
M run/create_p3_lookupTable_2-v5.2.3.f90 --> offline lookup table generation code 
D run/p3_lookupTable_1.dat-5.3-3momI.gz 

TESTS CONDUCTED: 
1. The mods fix the problems identified. 
2. Extensive testing in idealized supercell case and real cases in Canadian GEM model.
3. Jenkins tests are all pass.

RELEASE NOTE: Updates to 4 options of the P3 scheme (MP options 50, 51, 52, 53). The 3-moment lookup table file size is reduced by an order of magnitude (file names are changed for the lookup tables).  Improved flexibility has been added by including Kogan (2013, JAS) option for cloud autoconversion/accretion. A few minor bug fixes to harden the code have been added. References are:

Morrison, H., and J. A. Milbrandt, 2015: Parameterization of cloud microphysics based on the prediction of bulk particle properties. Part 1: Scheme description and idealized tests. J. Atmos. Sci., 72, 287-311.

Milbrandt, J. A., and H. Morrison, 2016: Parameterization of cloud microphysics based on the prediction of bulk particle properties. Part 3: Introduction of multiple free categories. J. Atmos. Sci., 73, 975-995.

Milbrandt, J. A., H. Morrison, D. T. Dawson II, and M. Paukert, 2021: A triple-moment representation of ice in the Predicted Particle Properties (P3) microphysics scheme. J. Atmos. Sci., 439-458.

### CODE REVIEW

**Comment by davegill** [comment]
@hmorrison100 
THANK YOU FOR GETTING THIS TO US SO FAST!

**Comment by davegill** [comment]
@weiwangncar @dudhia 
This seems an easy one to review. Mostly updates to table files, and a few code-hardening bug fixes

**Comment by weiwangncar** [comment]
@hmorrison100 Do you want to add something for the RELEASE NOTE part of the PR message?

**Comment by hmorrison100** [comment]
@weiwangncar Hi Wei, I modified the PR to add some info for "RELEASE NOTE". Thanks.

**Comment by davegill** [comment]
@dudhia @weiwangncar 
Folks,
This one looks ready now. Hugh added information to the RELEASE NOTES area. I copied down some of the description to fill it out.

**Review by weiwangncar — APPROVED** [review]

File: phys/module_mp_p3.F [hunk 1]
```diff
@@ -19,8 +19,8 @@
 !    Jason Milbrandt (jason.milbrandt@canada.ca)                                           !
 !__________________________________________________________________________________________!
 !                                                                                          !
-! Version:       4.2.3                                                                     !
-! Last updated:  2021-SEP                                                                  !
+! Version:       4.3.3                                                                     !
```
Comment (dudhia) [inline phys/module_mp_p3.F:22]: are you sure you don't want this to be 4.4?

File: phys/module_mp_p3.F [hunk 2]
```diff
@@ -118,10 +118,10 @@ subroutine p3_init(lookup_file_dir,nCat,trplMomI,model,stat,abort_on_err)
 
 ! Local variables and parameters:
  logical, save                  :: is_init = .false.
- character(len=1024), parameter :: version_p3                    = '4.2.3' 
+ character(len=1024), parameter :: version_p3                    = '4.3.3'
```
Comment (dudhia) [inline phys/module_mp_p3.F:121]: 4.4?

**Review by dudhia — APPROVED** [review]

**Comment by hmorrison100** [comment]
@dudhia The version number 4.3.3 is actually the internal P3 versioning, not related to WRF. Just by coincidence it's 4.3.3 like the current WRF.

**Comment by huangynj** [comment]
@hmorrison100 Hi Hugh, 'qcacc' in line 3469 should be 'qccon'
https://github.com/wrf-model/WRF/blob/a49fe581285d141ad8e9b2fce65192b194631a27/phys/module_mp_p3.F#L3469 
--- Yongjie

**Comment by hmorrison100** [comment]
@huangynj Hi Yongjie, thanks for pointing this out. We (Melissa) actually found this bug about 4 weeks ago. The impact is negligible so the plan is to fix this but it's not urgent.

**Pull request merged**